source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -0,0 +1,139 @@
+
+### Upload OVA generated by an external source
+If the OVA was generated by external source, the rest of the process will be:  
+
+1. The back-end adds the VM entity into the database.
+2. The back-end creates a temporary volume to store the OVA in.
+3. The back-end requests (through VDSM) the imagoio proxy to prepare to upload-image-like process.
+4. The proxy returns a URL (similar to the upload-image flow).
+5. The back-end passes this URL to the front-end (similar to the upload-image flow).
+6. The front-end starts transmitting the **whole OVA**.
+7. The proxy streams the input into the created volume.
+8. When the streaming is done, the back-end triggers the existing convert OVA method.
+9. The back-end asks to remove the temporary volume upon completion.
+
+Note that this is a second-priority task.  
+
+## Export as OVA
+The user will be able to specify a VM or a template to be exported as an OVA. In that case, the user needs to provide a path on a host where the OVA file is going to be created. The export process will then be comprised of the following operations:  
+
+1. The front-end passes the back-end a VM (or template) to export and a path on a host that the OVA should be created in.
+2. Each of the disks of the VM/template that either contains more than one volume or whose volume is not raw will be collapsed and converted to raw at some storage domain (TBD: sparsed?).
+3. A raw volume for the target OVA will be created in the specified path on the specified host. Its size would be the sum of the actual size of the VM disks + size for headers + size of OVF.
+4. The back-end produces the configuration of the entity to export in the form of an OVF.
+3. VDSM writes the OVF and the content of the disks according to the OVA specification to the OVA volume.
+4. The back-end asks to remove the temporary disk volumes upon completion.
+
+## Enhance Import OVA
+
+### Import OVA generated by oVirt
+Importing an OVA that was generated by oVirt won't involve virt-v2v and will be similar to the upload-ova process:  
+
+1. VDSM extracts the OVF from the OVA.
+2. The front-end allows the user to configure the parameters for the import.
+3. The back-end adds the VM(/template) entity into the database.
+4. The back-end asks VDSM to create volumes for the virtual disks.
+5. VDSM requests imageio proxy to start upload-OVA with the relevant parameters.
+6. VDSM starts sending the content of the disks from the OVA to the proxy which in turn streams it into the corresponding volumes.
+
+### Listing several OVA files within a directory
+It will be possible to trigger mass import of OVA files that reside within the same directory:
+
+1. The front-end passes a path to a directory on a host.
+2. VDSM iterates the OVA files in the given directory and extracts the OVF files within them.
```

Comment:
I share the feeling about vdsm as a project and the desire to avoid/bypass changes there when we can, but unfortunately it is our agent on the host. how do you suggest to do this differently?